.github/workflows/pr-title.yml

```diff
@@ -0,0 +1,18 @@
+# Validates that PR titles follow conventional commits
+name: "PR Title Lint"
+
+on:
+  pull_request_target:
+    types:
+      - opened
+      - edited
+      - synchronize
+
+jobs:
+  main:
+    name: Validate PR title
+    runs-on: ubuntu-latest
+    steps:
+      - uses: amannn/action-semantic-pull-request@v4
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
This line is probably the kicker for AppSec review. Otherwise, I'd say we should just review the code. We should also probably use the SHA hash of the action version instead of `@version` which will help our security posture.